modules/assim.sequential/R/Analysis_sda.R

```diff
@@ -151,128 +153,170 @@ GEF<-function(settings, Forecast, Observed, H, extraArg, nitr=50000, nburnin=100
-                                     name = 'space')
-    ## Adding X.mod,q,r as data for building model.
-    conf_tobit2space <<- configureMCMC(tobit2space_pred, thin = 10, print=TRUE)
-    conf_tobit2space$addMonitors(c("pf", "muf","y.censored")) 
-    ## [1] conjugate_dmnorm_dmnorm sampler: X[1:5]
-    ## important!
-    ## this is needed for correct indexing later
-    samplerNumberOffset_tobit2space <<- length(conf_tobit2space$getSamplers())
     
+    #### These vectors are used to categorize data based on censoring from the interval matrix
+    x.ind <- x.censored <- matrix(NA, ncol=ncol(X), nrow=nrow(X))
     for(j in seq_along(mu.f)){
       for(n in seq_len(nrow(X))){
-        node <- paste0('y.censored[',n,',',j,']')
-        conf_tobit2space$addSampler(node, 'toggle', control=list(type='RW'))
-        ## could instead use slice samplers, or any combination thereof, e.g.:
-        ##conf$addSampler(node, 'toggle', control=list(type='slice'))
+        x.ind[n,j] <- as.numeric(X[n,j] > 0)
+        x.censored[n,j] <- as.numeric(ifelse(X[n,j] > intervalX[j,2], 0, X[n,j])) #
       }
     }
     
-    #conf_tobit2space$printSamplers()
-    
-    Rmcmc_tobit2space <<- buildMCMC(conf_tobit2space)
+    recompileTobit <- extraArg$recompileTobit
+    recompileGEF <- extraArg$recompileGEF
     
-    Cmodel_tobit2space <<- compileNimble(tobit2space_pred)
-    Cmcmc_tobit2space <<- compileNimble(Rmcmc_tobit2space, project = tobit2space_pred)
+    ###TO DO: needs to recompile if there are new data sources
+    if(TRUE){ #any(X==0,na.rm=T)
+      #
+    if(t == 1 | recompileTobit){
+      #The purpose of this step is to impute data for mu.f 
+      #where there are zero values so that 
+      #mu.f is in 'tobit space' in the full model
+      constants.tobit2space <<- list(N = nrow(X),
+                                     J = length(mu.f))
+      
+      data.tobit2space <<- list(y.ind = x.ind,
+                                y.censored = x.censored,
+                                mu_0 = rep(0,length(mu.f)),
+                                lambda_0 = diag(1000,length(mu.f)), #can try solve
+                                nu_0 = ncol(X)+1,
+                                wts = wts*nrow(X), #sigma x2 max Y
+                                Sigma_0 = solve(diag(1000,length(mu.f))))#some measure of prior obs
+      
+      inits.tobit2space <- function() list(muf = rmnorm_chol(1,colMeans(X),chol(diag(ncol(X))*100)),
+                                  pf = rwish_chol(1,df = ncol(X)+1,cholesky = chol(solve(cov(X)))))
+      
+      #ptm <- proc.time()
+      tobit2space_pred <<- nimbleModel(tobit2space.model, data = data.tobit2space,
+                                       constants = constants.tobit2space, inits = inits.tobit2space(),
+                                       name = 'space')
+      
+      logprob_y_tobit2space <- tobit2space_pred$calculate('y.censored')
```

Comment:
I get `NA` here I don't know why. At first I thought It's because it's called before adding it as monitor in L208, but even if I add that first it's `NA` and breals the next line in the `if`. Is there something I'm missing ?

Reply:
it turns out, it was bc of my weights were `NA`, If I choose a series of weights this will return some number. Maybe an `if` would help here.